department-of-veterans-affairs/caseflow
# Add CavcDashboard components and react route

Resolves [APPEALS-13231](https://vajira.max.gov/browse/APPEALS-13231)

### Description
- Added route to config/routes.rb for "/queue/appeals/:appealId/cavc_dashboard"
- Added route to QueueApp for "/queue/appeals/:appealId/cavc_dashboard"
- Added CavcDashboard component
- Added CavcDashboardLoadingScreen component
- Added feature tests for CavcDashboard component
- Added trait :oai_user to User factory for future tests when dashboard authorization by role is implemented

### Acceptance Criteria
- [ ] For a CAVC Remand appeal, the CAVC Dashboard placeholders render
- [ ] For a non-CAVC appeal, the CAVC Dashboard redirects to the Case Details page

### Testing Plan
1. Go to demo environment
2. Go to Search
3. Search for a CAVC Remand by veteran file number (200000250 - 200000260 should have CAVC Remands)
4. Go to case details page for that appeal
5. Once at the case details page, add "/cavc_dashboard" to the URL and go to that page
6. Verify that the page loads with placeholder text displaying the appeal UUID (from the URL)
7. Go back to Search
8. Search for a case that is NOT a CAVC remand (100000001 - 100000010 should not have CAVC remands)
9. Go to case details page for that appeal
10. Once at the case details page, add "/cavc_dashboard" to the URL and go to that page
11. Verify that the page redirects back to case details without loading the placeholder text, and that the URL does not contain "/cavc_dashboard" 

### Code Documentation Updates
